cartridge.lua

```diff
@@ -730,6 +730,27 @@ local function cfg(opts, box_opts)
         return nil, err
     end
 
+    -- Stop roles on shutdown
+    if box.ctl.on_shutdown ~= nil then
+        box.ctl.on_shutdown(function()
+            local all_roles = roles.get_all_roles()
+            local opts = {is_master = require('cartridge.failover').is_leader()}
+
+            for _, role_name in pairs(all_roles) do
+                if service_registry.get(role_name) ~= nil then
+                    local role = roles.get_role(role_name)
+
+                    if type(role.stop) == 'function' then
+                        pcall(role.stop, opts)
```

Comment:
I suppose you need to write the logs in case of an error

Reply:
Done